src/window.rs

```diff
@@ -0,0 +1,138 @@
+use core::marker::PhantomData;
+use core::f64;
+
+use {Sample, FloatSample};
+use signal;
+use signal::{Phase, ConstHz, Step};
+use conv::{FromSample, ToSample};
+use frame::Frame;
+
+// Using types instead of enum allows for static dispatch
+pub trait WindowType<S: Sample, T: Step> {
```

Comment:
Also, I think we could do the following renamings:

`WindowType` -> `Type`
`HanningWindow` -> `Hanning`
`RectangleWindow` -> `Rectangle`

as each are already within the `window` namespace, i.e. `window::Type`, `window::Hanning`, `window::Rectangle`.